### DISCUSSION TITLE
Should results for a given string change over time?

### DISCUSSION THREAD

**Discussion by spartanhooah** [adirh3/Fluent-Search#413] [Q&A]
I've been trying this out for the last couple months and generally liking it as a replacement for Keypirinha. However, I would expect such a tool to adapt to my queries over time, and in some cases it has (e.g. "pa" resolves "PatchMyPc.exe" before "App Paint"). However, in some cases this isn't working. For whatever reason, I had to manually add the location where the IntelliJ shortcut from the Start Menu is. After opening it several times via Fluent Search, the query "int" still has "File IntelliJ IDEA Community Edition 2021.2.lnk" in the 9th slot. How can I get this to bubble up higher over time?

**Comment by adirh3** [comment]
Hey @spartanhooah, thanks for the feedback!
The part you're talking about is the machine learning feature of Fluent Search, which is not as good as I'd like it to be.
In the recent nightly versions I have tried to improve, can you please update to it and let me know if you notice any difference?  (Updating is done by changing Settings -> System -> Release feed -> Nightly)
Other than machine learning, Fluent Search offers more way to prioritize results -
1. You can pin the result to show on top using the 📌 at top right or pressing Alt+P
2. You can prioritize results based on type or file path in Settings (advanced) -> Prioritization. It's buggy in the stable version, so I suggest using nightly for it if you try it.
Let me know if this helps, and feel free to keep sharing any feedback/questions here or in Discord.
